include/fiction/algorithms/simulation/sidb/sidb_simulation_engine.hpp

```diff
@@ -41,8 +49,77 @@ enum class exhaustive_sidb_simulation_engine
      * *QuickExact* is also an exact simulation engine that requires exponential runtime, but it scales a lot better
      * than ExGS due to its effective search-space pruning.
      */
-    QUICKEXACT
+    QUICKEXACT,
+    /**
+     * *ClusterComplete* is a novel exact simulation engine that requires sub-exponential runtime, for simulation
+     * problems in practice like SiDB logic. It handles base 3 simulation very effectively, since the engine makes no
+     * distinction in the simulation methodology for a different base number.
+     */
+    CLUSTERCOMPLETE
 };
+/**
+ * Selector exclusively for heuristic SiDB simulation engines.
+ */
+enum class heuristic_sidb_simulation_engine
+{
+    /**
+     * *QuickSim* is a heuristic simulation engine that only requires polynomial runtime.
+     */
+    QUICKSIM
+};
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wswitch"
+// @ql:off
+/**
+ * Returns the name of the given simulation engine.
+ *
+ * @tparam EngineType The type of the sidb engine (exhaustive/heuristic/general).
+ * @param sidb_simulation_engine A SiDBs simulation engine.
```

Comment:
```suggestion
 * @param engine A SiDBs simulation engine.
```

Reply:
This name conflicts with the name of the `enum`

Reply:
Apologies, I misread. Thanks!